superset/migrations/versions/78ee127d0d1d_reconvert_legacy_filters_into_adhoc.py

```diff
@@ -0,0 +1,72 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""reconvert legacy filters into adhoc
+
+Revision ID: 78ee127d0d1d
+Revises: c2acd2cf3df2
+Create Date: 2019-11-06 15:23:26.497876
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = "78ee127d0d1d"
+down_revision = "c2acd2cf3df2"
+
+import json
+import logging
+import uuid
+from collections import defaultdict
+
+from alembic import op
+from sqlalchemy import Column, Integer, Text
+from sqlalchemy.ext.declarative import declarative_base
+
+from superset import db
+from superset.utils.core import (
+    convert_legacy_filters_into_adhoc,
+    split_adhoc_filters_into_base_filters,
+)
+
+Base = declarative_base()
+
+
+class Slice(Base):
+    __tablename__ = "slices"
+
+    id = Column(Integer, primary_key=True)
+    params = Column(Text)
+
+
+def upgrade():
+    bind = op.get_bind()
+    session = db.Session(bind=bind)
+
+    for slc in session.query(Slice).all():
+        if slc.params:
+            try:
+                params = json.loads(slc.params)
+                convert_legacy_filters_into_adhoc(params)
+                slc.params = json.dumps(params, sort_keys=True)
```

Comment:
maybe only do this if the parameters are different? And it might make sense to do a session.commit() after each iteration instead of one big one at the end. Not sure how a db with 200k slices would handle this

Reply:
@etr2460 I can definitely make the change to only update the params if they differ which the ORM will track. In terms of the commit I tested this with Airbnb's production database which has ~ 200k records. Batching is preferred over single record commits.

Reply:
Chiming in here, I also prefer one big commit at the end to make sure the migration doesn't die half way through, leaving the backend half migrated. If it takes long then so be it. Re: the comment about only updating rows with changed params; definitely agree, and it should speed up the commit at the end.